fda/basis.py

```diff
@@ -2189,38 +2190,51 @@ def __getitem__(self, key):
 
         if isinstance(key, int):
             return self.copy(coefficients=self.coefficients[key:key + 1])
-
         else:
             return self.copy(coefficients=self.coefficients[key])
 
+    def plus_samples(self):
+        if self.nsamples == 1:
+            return self
+        return self[0] + (self[1:].plus_samples())
+
     def __add__(self, other):
         """Addition for FDataBasis object."""
+        if isinstance(other, FDataBasis):
+            if self.basis != other.basis:
+                raise NotImplementedError
+            else:
+                return FDataBasis(self.basis.copy(),
+                                  self.coefficients + other.coefficients)
 
-        raise NotImplementedError
+        coefs = self.coefficients.copy()
+        coefs[:, 0] = self.coefficients[:, 0] + numpy.array(other)
+        return FDataBasis(self.basis.copy(), coefs)
 
     def __radd__(self, other):
         """Addition for FDataBasis object."""
-
         return self.__add__(other)
 
     def __sub__(self, other):
         """Subtraction for FDataBasis object."""
-
-        raise NotImplementedError
+        if isinstance(other, list):
+            other = numpy.array(other)
+        return self.__add__(-1 * other)
 
     def __rsub__(self, other):
         """Right subtraction for FDataBasis object."""
-
-        raise NotImplementedError
+        return (-1 * self).__add__(other)
 
     def __mul__(self, other):
         """Multiplication for FDataBasis object."""
+        if isinstance(other, FDataBasis):
+            raise NotImplementedError
 
-        raise NotImplementedError
+        mult = numpy.atleast_2d(other).reshape(-1, 1)
```

Comment:
Why the reshape?

Reply:
Because a need a column vector to apply only on the first coordinate.